editoast/src/core/simulation.rs

```diff
@@ -66,29 +69,135 @@ pub struct PhysicsRollingStock {
     pub raise_pantograph_time: Option<u64>,
 }
 
-#[derive(Debug, Default)]
-pub struct SimulationParameters {
+#[derive(Debug, Clone, Default)]
+pub struct PhysicsConsist {
+    /// In kg
     pub total_mass: Option<f64>,
+    /// In m
     pub total_length: Option<f64>,
+    /// In m/s
     pub max_speed: Option<f64>,
+    pub towed_rolling_stock: Option<TowedRollingStock>,
 }
 
-impl PhysicsRollingStock {
-    pub fn new(traction_engine: RollingStock, params: SimulationParameters) -> Self {
-        let traction_engine_length = traction_engine.length * 1000.0;
-        let length = params
+impl PhysicsConsist {
+    pub fn compute_length(&self, traction_engine: &RollingStock) -> u64 {
+        let towed_rolling_stock_length = self
+            .towed_rolling_stock
+            .as_ref()
+            .map(|trs| trs.length)
+            .unwrap_or(0.0);
+
+        let length = self
             .total_length
-            .map(|tl| tl * 1000.0)
-            .unwrap_or(traction_engine_length)
-            .round() as u64;
+            .unwrap_or(traction_engine.length + towed_rolling_stock_length);
+
+        // Convert m to mm
+        let length = length * 1000.0;
+
+        length.round() as u64
+    }
 
+    pub fn compute_max_speed(&self, traction_engine: &RollingStock) -> f64 {
+        if let Some(max_speed_parameter) = self.max_speed {
+            f64::min(traction_engine.max_speed, max_speed_parameter)
+        } else {
+            traction_engine.max_speed
+        }
+    }
+
+    pub fn compute_startup_acceleration(&self, traction_engine: &RollingStock) -> f64 {
+        if let Some(towed_rolling_stock) = self.towed_rolling_stock.as_ref() {
+            f64::max(
+                traction_engine.startup_acceleration,
+                towed_rolling_stock.startup_acceleration,
+            )
+        } else {
+            traction_engine.startup_acceleration
+        }
+    }
+
+    pub fn compute_comfort_acceleration(&self, traction_engine: &RollingStock) -> f64 {
+        if let Some(towed_rolling_stock) = self.towed_rolling_stock.as_ref() {
+            f64::min(
+                traction_engine.comfort_acceleration,
+                towed_rolling_stock.comfort_acceleration,
+            )
+        } else {
+            traction_engine.comfort_acceleration
+        }
+    }
```

Comment:
The functional style is mostly used in this file, can these be rewritten in this style for consistency?